dm/worker/worker.go

```diff
@@ -165,119 +189,85 @@ func (w *Worker) Close() {
 	// cancel status output ticker and wait for return
 	w.cancel()
```

Comment:
the close sequence is close
`relayHolder` -> `relayPurger` -> `meta log` -> `worker it self`
we'd better to ensure the component closed later must not reference to object defined in closed earlier. If not ensured, the behavior may be unpredictable.
For example, currently in worker there is a `handleTask` background routine, in this routine `meta` and `relayPurger` may be used. I'm not sure whether this is the root case that causes the not exited state dm-worker.

Reply:
Anyway, you're right. I admire you

Reply:
I review the implementation, it's not luck to find the deadlock, but I still fix it and re-run integrated test